### ISSUE TITLE
The API version 2024-05-04 is not supported by Azurite. Please upgrade Azurite to latest version and retry. If you are using Azurite in Visual Studio, please check you have installed latest Visual Studio patch. Azurite command line parameter "--skipApiVersionCheck" or Visual Studio Code configuration "Skip Api Version Check" can skip this error. 

### ISSUE THREAD

**Issue by varndellwagglebee** [dotnet/aspire#4646]
Getting the following azurite error.  I upgrade Azurite and still the same message:
The API version 2024-05-04 is not supported by Azurite. Please upgrade Azurite to latest version and retry. If you are using Azurite in Visual Studio, please check you have installed latest Visual Studio patch. Azurite command line parameter "--skipApiVersionCheck" or Visual Studio Code configuration "Skip Api Version Check" can skip this error.

**Comment by JordanZaerr** [comment]
Ran into this today and had to poke around in their [sample app](https://learn.microsoft.com/en-us/dotnet/aspire/storage/azure-storage-components?tabs=visual-studio%2Cdotnet-cli&pivots=azurite) to figure out why it worked and my code didn't.  

If you target `Azure.Storage.Blobs` version `12.19.1` instead of `12.20.0` then it avoids this error. That package is a dependency of `Aspire.Azure.Storage.Blobs`. That fixed it for me at least.

**Comment by for-shariq** [comment]
Facing the same issue.

**Comment by JordanZaerr** [comment]
They released 8.0.2 after my previous comment.  I'm sure this will be updated eventually as well but they released a new version of Azurite that supports this API version([3.3.0](https://github.com/Azure/Azurite/releases/tag/v3.30.0)).

The `RunAsEmulator` call hardcodes version 3.29.0 inside that extension method.  So I just call that function again with the new version...

```csharp
storage.RunAsEmulator()
    .WithAnnotation(new ContainerImageAnnotation
    {
        Registry = "mcr.microsoft.com",
        Image = "azure-storage/azurite",
        Tag = "3.30.0"
    });
```

That should download the new azurite container and start it up as usual.

**Comment by VasilisPlavos** [comment]
As JordanZaerr said, just use Azure.Storage.Blobs version 12.19.1 and it will work.  😊

**Comment by zbarrier** [comment]
> They released 8.0.2 after my previous comment. I'm sure this will be updated eventually as well but they released a new version of Azurite that supports this API version([3.3.0](https://github.com/Azure/Azurite/releases/tag/v3.30.0)).
> 
> The `RunAsEmulator` call hardcodes version 3.29.0 inside that extension method. So I just call that function again with the new version...
> 
> ```cs
> storage.RunAsEmulator()
>     .WithAnnotation(new ContainerImageAnnotation
>     {
>         Registry = "mcr.microsoft.com",
>         Image = "azure-storage/azurite",
>         Tag = "3.30.0"
>     });
> ```
> 
> That should download the new azurite container and start it up as usual.

This worked with 8.0.2.  Thank you!

**Comment by JordanZaerr** [comment]
@zbarrier I edited my comment because in the PR I linked I found a cleaner way to do the same thing.

**Comment by napalm684** [comment]
FYI with the release of 8.1.0 today you need to use an Azurite docker container with the 3.31.0 tag.
```
var storage = builder.AddAzureStorage("storage").RunAsEmulator(c => c.WithImageTag("3.31.0"))
```

**Comment by shivamverma-ms** [comment]
Hi Folks, with the new GA version of Azure Storage 2.0.0, it is only compliant with 12.20.0 version of Azure.Storage.Blobs, in this version the mismatch issue is not getting resolved. What should we do here?

Added to this, we need to use blob uri of azurite instead of connection string, as part of the new methods introduced in azure storage 2.0.0

**Comment by kwaazaar** [comment]
Yep, issue is back. I "fixed" it by referencing (=not updating) the Aspire.Azure.Storage.Blobs package to 8.2.0. By keeping it at 8.0.2 you can also keep using Azure.Storage.Blobs 12.20.0.

**Comment by hansmbakker** [comment]
When using these components:
- `Aspire.Azure.Storage.Blobs` / `Aspire.Hosting.Azure.Storage` version `8.2.1`
- docker image `azure-storage/azurite:3.31.0`

The error is now
> The API version 2024-11-04 is not supported by Azurite

In my opinion it is a new symptom of the same issue; if you feel it should be a new issue, please convert this comment into an issue.

**Comment by davidfowl** [comment]
@sebastienros can you look at this?

**Comment by kwaazaar** [comment]
Why not use the 'latest' image tag? Solved the issues for me.

```
        var blobs = builder.AddAzureStorage("storage")
            .RunAsEmulator(c => c
                .WithDataVolume()
                **.WithImageTag("latest")**
             )
            .AddBlobs("blobs");

```

**Comment by sebastienros** [comment]
I have [submitted a PR](https://github.com/dotnet/aspire/pull/7239) to add `--skipApiVersionCheck` by default. Thanks for testing it.

**Comment by davidfowl** [comment]
This is fixed in 9.1, in ~an hour so you can try a build here and let us know if we missed anything https://github.com/dotnet/aspire/blob/main/docs/using-latest-daily.md 😄
